api/network/v1alpha1/interface.zip.go

```diff
@@ -25,8 +25,13 @@ type NetworkInterfaceSpec struct {
 	// The name of the interface.
 	IfName string `json:"ifname,omitempty"`
 
-	// IP address of a machine interface.
-	IP string `json:"ip,omitempty"`
+	// Fields used to define a NetdevIp.
+	CIDR     string
+	Gateway  string
+	DNS0     string
+	DNS1     string
+	Hostname string
+	Domain   string
```

Comment:
Can you add field names:

```Go
// IPv4 address in CIDR notation, which includes the subnet.
CIDR string

// Gateway IPv4 address.
Gateway string

// IPv4 address of the primary DNS server.
DNS0 string

// IPv4 address of the secondary DNS server.
DNS1 string

// Hostname of the IPv4 address.
Hostname string

// Domain/Search suffix for IPv4 address.
Domain string
```
From: https://github.com/unikraft/unikraft/blob/9ff29304e91534e8d05ae27533b995d74fb09ec2/lib/uknetdev/include/uk/netdev_core.h#L224-L238

Reply:
Added